great_expectations/compatibility/aws.py

```diff
@@ -3,21 +3,20 @@
 from great_expectations.compatibility.not_imported import NotImported
 
 AWS_NOT_IMPORTED = NotImported(
-    "Unable to load AWS connection object; please 'pip install boto3 botocore'"
+    "AWS connection components are not installed, please 'pip install boto3 botocore'"
 )
 
 try:
-    import boto3
+    import boto3  # noqa TID251
 except ImportError:
     boto3 = AWS_NOT_IMPORTED
 
 try:
-    import botocore
+    import botocore  # TID251
```

Comment:
I'm seeing this one as ['banned API`](https://beta.ruff.rs/docs/rules/banned-api/). Is there a reason why we weren't seeing this before?

Reply:
@Shinnnyshinshin I am still trying to figure out how to avoid this `# TID251`.  Thanks.

Reply:
Hmmm interesting.  Maybe @Kilo59 could weigh in on this... it just feels a bit wrong to override this

Reply:
@alexsherstinsky it looks like there are other places in the codebase that have this flag, including a few that were introduced by this PR. `https://github.com/great-expectations/great_expectations/pull/8123/` for sqlalchemy. I think we are ok to go here for now